### PR TITLE
Add standard release workflows

### LINKED ISSUE #65: One click release
When the time comes, one click / action should be all we need to do create a "release" that provides us with download-able production binaries.

### PR DESCRIPTION
**Pull request by thomaseizinger**
These are mostly copy-pasted from xmr-btc-swap modulo the changelog
handling (because we don't have a changelog).

The gist is:

- We attach release binaries to every release that is created.
- Every merge to master creates a `preview` release.
- A merge from a branch that is named `release/x.y.z` creates a
release `x.y.z`.
- A new release branch can be created via `draft-new-release.yml`.

Fixes #65.

Example runs of these workflows can be found here: https://github.com/thomaseizinger/hermes/actions

### CODE REVIEW

File: .github/workflows/build-release-binary.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+      - name: Install compiler for armhf arch
+        if: matrix.target == 'armv7-unknown-linux-gnueabihf'
+        run: |
+          sudo apt-get update
+          sudo apt-get install gcc-arm-linux-gnueabihf
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
```
Comment (bonomat) [inline .github/workflows/build-release-binary.yml]: Any particular reason why version 14? 
I'm on 16.2

**Review by bonomat — APPROVED** [review]
Thank you. 
The actions still mention `changelog` here and there but that's not an issue.

File: .github/workflows/build-release-binary.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/build-release-binary.yml]: Nope, just didn't know there was already 16!

**Review by da-kami — APPROVED** [review]

**Review by bonomat — APPROVED** [review]
Looks good

File: .github/workflows/build-release-binary.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+jobs:
+  build_binaries:
+    name: Build release binary
+    strategy:
+      matrix:
+        include:
+          - bin: taker
+            target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+            archive_ext: tar
```
Comment (klochowicz) [inline .github/workflows/build-release-binary.yml:16]: out of curiosity: why `.tar` and not more conventional `.tar.gz`?

File: .github/workflows/build-release-binary.yml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          - bin: taker
+            target: armv7-unknown-linux-gnueabihf
+            os: ubuntu-latest
+            archive_ext: tar
+          - bin: taker
+            target: x86_64-apple-darwin
+            os: macos-latest
+            archive_ext: tar
```
Comment (klochowicz) [inline .github/workflows/build-release-binary.yml:24]: on macos `zip` is probably more convenient choice, as it automatically unpacked by Safari when extracted

**Review by klochowicz — APPROVED** [review]

File: .github/workflows/build-release-binary.yml (hunk 2 above)
Comment (thomaseizinger) [reply .github/workflows/build-release-binary.yml:16]: We are not compressing things so it is just an archive!

File: .github/workflows/build-release-binary.yml (hunk 3 above)
Comment (thomaseizinger) [reply .github/workflows/build-release-binary.yml:24]: Do you mind sending a follow-up patch if you care enough? :D

Testing Github workflows is a pain ...

Comment (klochowicz) [reply .github/workflows/build-release-binary.yml:24]: I'll look into it at some point, when all the MVP tasks are done :)

**Pull request merged**